bindings/go/runtime/registry.go

```diff
@@ -22,46 +24,101 @@ type Scheme struct {
 	// if the constructors cannot determine a match,
 	// this will trigger the creation of an unstructured.Unstructured with NewScheme instead of failing.
 	allowUnknown bool
-	aliases      map[Type]Type
-	defaults     map[Type]Typed
+	// defaults maps default Types to their reflect.Type.
+	// the reflect.Type is the authoritative type for the default Type.
+	defaults *bimap.Map[Type, reflect.Type]
+	// aliases maps alias Types to their default Type.
+	// they are alternative forms of the default Type.
+	aliases map[Type]Type
+	// instances maps reflect.Type to their prototype instance.
+	// this avoids the need to create new instances via reflection and allows
+	// passing pre-initialized default values if needed.
+	instances map[reflect.Type]Typed
+}
+
+// NewScheme creates a new registry.
+func NewScheme(opts ...SchemeOption) *Scheme {
+	reg := &Scheme{
+		defaults:  bimap.New[Type, reflect.Type](),
+		aliases:   map[Type]Type{},
+		instances: make(map[reflect.Type]Typed),
+	}
+	for _, opt := range opts {
+		opt(reg)
+	}
+	return reg
+}
+
+// GetTypes returns a map of all registered types, where the keys are the default types
+// and the values are slices containing all aliases for that type, sorted lexicographically.
+func (r *Scheme) GetTypes() map[Type][]Type {
+	result := map[Type][]Type{}
+	for k, v := range r.GetTypesIter() {
+		result[k] = slices.SortedFunc(v, CompareTypesLexicographically)
+	}
+	return result
 }
 
-// GetTypes returns a map of all registered types.
-// The keys are the default types, and the values are slices containing all aliases for that type.
+// GetTypesIter returns an iterator of all registered types.
+// The keys are the default types, and the values are iterators containing all aliases for that type.
 // If a type has no aliases, it will have an empty slice as its value.
 // The slices of aliases are always sorted for consistent ordering.
-func (r *Scheme) GetTypes() map[Type][]Type {
-	r.mu.RLock()
-	defer r.mu.RUnlock()
+func (r *Scheme) GetTypesIter() iter.Seq2[Type, iter.Seq[Type]] {
+	return func(yield func(Type, iter.Seq[Type]) bool) {
+		r.mu.RLock()
+		defer r.mu.RUnlock()
+		for typ := range r.defaults.Iter() {
+			if !yield(typ, r.AliasesIter(typ)) {
```

Comment:
Does this double lock the same mutex? 🤔  getTypesIter r.mu.RLock then `AliasesIter` r.mu.RLock's the same mutex again.

Reply:
Nah I think this is okay. In either case it should return. The thing I was wondering about is if it should preload items rather then lock. Because if people take too long to process items during iteration the read lock is held during the FULL iteration. And no write cna happen until the iterator is done.

But I think that's a premature optimize that isn't needed. And I have no data to back it up. :)